activerecord/lib/active_record/database_configurations.rb

```diff
@@ -57,7 +57,7 @@ def configs_for(env_name: nil, spec_name: nil, include_replicas: false)
     # return the first config hash for the environment.
     #
     #   { database: "my_db", adapter: "mysql2" }
-    def default_hash(env = ActiveRecord::ConnectionHandling::DEFAULT_ENV.call.to_s)
+    def default_hash(env = ActiveRecord::Base.current_environment)
```

Comment:
This is the logic for `current_environment` method: `(Rails.env if defined?(Rails.env)) || ENV["RAILS_ENV"].presence || ENV["RACK_ENV"].presence  || "default_env"`

It seems like it will always return a string so the `to_s` call is redundant.